e2e/test/prerequisites/E2ETestsSetup/e2eTestsSetup.ps1

```diff
@@ -360,15 +399,19 @@ if (-not $spExists) {
 # Adding retries so that we can grant the permissions successfully without re-running the script.
 $retry = $true
 $retryCount = 1;
-while ($retry -eq $true) {
-    try {
+while ($retry)
+{
+    try
+    {
         Write-Host "`nGranting $appId Reader role assignment to the $Resourcegroup resource group - Trial $retryCount"
         az role assignment create --role Reader --assignee $appId --resource-group $ResourceGroup --output none
         $retry = $false
     }
-    catch {
+    catch
+    {
         $retryCount++
-        if ($retryCount -eq 10) {
+        if ($retryCount -ge 10)
+        {
```

Comment:
What kind of failure causes this to require retry? Should we sleep between retries? If max retries is reached, should the script be allowed to continue?

Reply:
The script would fail on error. We don't continue on Error (Set at the beginning of the script). Sleep might not be a bad idea to add but I don't know how much to add. I wanted everyone to try and see and then add it accordingly. I occasionally hit the issue that we created the SP but it said not found when we tried to assign RBAC (I'm guessing we hit a different endpoint where changes were not propagated).

Reply:
I don't see what causes the error to occur. We have a catch. If max retries reached, we exit the loop. What is throwing the error though?

Reply:
I see what you are saying. We need to error it out to hint the user to run the script again. If we continue without failing, user can assume everything is done but some E2E tests might fail as the permissions were not assigned.